### PR TITLE
Updated all components to latest version

### PR DESCRIPTION
**Pull request by soarqin**
Updated GCC to 7.1.0
Updated binutils to 2.28 to remove dirty patch
Updated GMP to 6.1.2 and its download url to avoid issue caused by site down
Updated ISL to 0.18
Updated libzip to 1.2.0

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -135,8 +135,8 @@ function(toolchain_deps toolchain_deps_dir toolchain_install_dir toolchain_suffi
 
     ExternalProject_Add(libzip${suffix}
         DEPENDS zlib${suffix}
-        URL https://web.archive.org/web/20170404153226/https://nih.at/libzip/libzip-1.1.3.tar.gz
-        URL_HASH SHA256=1faa5a524dd4a12c43b6344e618edce1bf8050dfdb9d0f73f3cc826929a002b0
+        URL https://nih.at/libzip/libzip-1.2.0.tar.gz
```
Comment (devnoname120) [inline CMakeLists.txt]: Downloading from nih.at from Travis-CI has a good chance of failing as past experience has shown.
Please use an alternative download link such as https://github.com/devnoname120/libzip/releases/download/1.2.0/libzip-1.2.0.tar.gz

**Comment by devnoname120** [comment]
Have you tested the updated components to make sure that they don't introduce new issues?

**Comment by soarqin** [comment]
@devnoname120 Since there is no test case, I have just tested building VitaShell and it worked fine.

**Comment by devnoname120** [comment]
Could you compile Vitashell and make sure that it still works properly? (VPK installation, etc.)

**Comment by devnoname120** [comment]
Could you try to installed this compiled version of Vitashell and see if installing a VPK works?

**Comment by soarqin** [comment]
For the sake of security, I recompiled VitaSDK from scratch, and built VitaShell and its dependencies from new compiled VitaSDK.
Then I tested installing VitaShell from MolecularShell and installing other vpks from new installed VitaShell, all worked fine.

**Comment by devnoname120** [comment]
OK, I will leave other vitasdk members to review the changes as I'm not familiar with this part of the SDK.

**Comment by soarqin** [comment]
RetroArch is probably the most complicate custom app in my knowledge and its authors have reported several critical bugs for VitaSDK due to large codebase. So I think that test RetroArch's build&run is enough to check if there is any issue imported.

**Comment by soarqin** [comment]
#45 includes all changes of this PR, close it

**Pull request closed without merging**